test/FlexVotingClient.t.sol

```diff
@@ -102,6 +102,250 @@ contract Deployment is FlexVotingClientTest {
+  }
+
+  function testFuzz_IncreasesOnDeposit(address _user, uint208 _amount) public {
+    _commonUserAssumptions(_user);
+    _amount = uint208(bound(_amount, 1, type(uint128).max));
+
+    // Deposit some gov.
+    _mintGovAndDepositIntoFlexClient(_user, _amount);
+
+    assertEq(flexClient.exposed_rawBalanceOf(_user), _amount);
+  }
+
+  function testFuzz_DecreasesOnWithdrawal(address _user, uint208 _amount) public {
+    _commonUserAssumptions(_user);
+    _amount = uint208(bound(_amount, 1, type(uint128).max));
+
+    // Deposit some gov.
+    _mintGovAndDepositIntoFlexClient(_user, _amount);
+
+    assertEq(flexClient.exposed_rawBalanceOf(_user), _amount);
+
+    vm.prank(_user);
+    flexClient.withdraw(_amount);
+    assertEq(flexClient.exposed_rawBalanceOf(_user), 0);
+  }
+
+  function testFuzz_UnaffectedByBorrow(address _user, uint208 _deposit, uint208 _borrow) public {
+    _commonUserAssumptions(_user);
+    _deposit = uint208(bound(_deposit, 1, type(uint128).max));
+    _borrow = uint208(bound(_borrow, 1, _deposit));
+
+    // Deposit some gov.
+    _mintGovAndDepositIntoFlexClient(_user, _deposit);
+
+    assertEq(flexClient.exposed_rawBalanceOf(_user), _deposit);
+
+    vm.prank(_user);
+    flexClient.borrow(_borrow);
+
+    // Raw balance is unchanged.
+    assertEq(flexClient.exposed_rawBalanceOf(_user), _deposit);
+  }
+}
+
+// Contract name has a leading underscore for scopelint spec support.
+contract _CastVoteReasonString is FlexVotingClientTest {
+  function test_ReturnsDescriptiveString() public {
+    assertEq(
+      flexClient.exposed_castVoteReasonString(), "rolled-up vote from governance token holders"
+    );
+  }
+}
+
+// Contract name has a leading underscore for scopelint spec support.
+contract _SelfDelegate is FlexVotingClientTest {
+  function testFuzz_SetsClientAsTheDelegate(address _delegatee) public {
+    vm.assume(_delegatee != address(0));
+    vm.assume(_delegatee != address(flexClient));
+
+    // We self-delegate in the constructor, so we need to first un-delegate for
+    // this test to be meaningful.
+    vm.prank(address(flexClient));
+    token.delegate(_delegatee);
+    assertEq(token.delegates(address(flexClient)), _delegatee);
+
+    flexClient.exposed_selfDelegate();
+    assertEq(token.delegates(address(flexClient)), address(flexClient));
+  }
+}
+
+// Contract name has a leading underscore for scopelint spec support.
+contract _CheckpointRawBalanceOf is FlexVotingClientTest {
+  function testFuzz_StoresTheRawBalanceWithTheBlockNumber(
+    address _user,
+    uint208 _amount,
+    uint48 _blockNum
+  ) public {
+    vm.assume(_user != address(flexClient));
+    vm.assume(_blockNum > 2);
+    vm.assume(_blockNum < type(uint48).max);
+    _amount = uint208(bound(_amount, 1, type(uint128).max));
+
+    flexClient.exposed_setDeposits(_user, _amount);
+    assertEq(flexClient.getPastRawBalance(_user, _blockNum), 0);
+
+    vm.roll(_blockNum);
```

Comment:
It would be best to do `vm.roll(block.number + _blockNum)` here, or alternatively, we can bound the `_blockNum` value to something above `block.number` in the first place, as suggested above.

Reply:
Taken care of in 4d6d8b2